innovation.game.php

```diff
@@ -1234,6 +1237,38 @@ function transferCardFromTo($card, $owner_to, $location_to, $bottom_to=false, $s
         }
         return $card;
     }
+
+    function setCardIconHash() {
+        //LMF: unsure if I have to grab this list from the db, I don't fully understand texual_card_infos.
+        $cards = self::getObjectListFromDB("
+            SELECT
+                id,
+                spot_1,
+                spot_2,
+                spot_3,
+                spot_4
+            FROM
+                card
+        ");
+        
+        foreach ($cards as $card) {
+            //icon_hash_key to be used instead of the db value for icons. 1 is used for hex icons, thus can be ignored through the product
+            $icon_hash_key = array(1,2,3,5,7,13,17);
+            $ind = $card['spot_1'];
```

Comment:
left over debug code. 

I was having an issue with null spots. figuring that out that they are possible means I can set them to 1, as you can see in the ternery operator. In the future the is_null/retrieving from the hash_key could be it's own function as we complicate things with echo/inspire/bonus

Reply:
In the meantime, there's a shorthand we can do with the ternary operator, I think. https://stackoverflow.com/questions/4998178/if-null-use-other-variable-in-one-line-in-php

Want to try this?

$icon_hash_key[$card['spot_1') ?: 0]